homeassistant/components/somfy/cover.py

```diff
@@ -1,49 +1,60 @@
 """Support for Somfy Covers."""
+
 from pymfy.api.devices.blind import Blind
 from pymfy.api.devices.category import Category
+import requests
 
 from homeassistant.components.cover import (
     ATTR_POSITION,
     ATTR_TILT_POSITION,
+    DEVICE_CLASS_BLIND,
+    DEVICE_CLASS_SHUTTER,
     CoverEntity,
 )
+from homeassistant.const import STATE_CLOSED, STATE_OPEN
+from homeassistant.helpers.restore_state import RestoreEntity
 
 from . import API, CONF_OPTIMISTIC, DEVICES, DOMAIN, SomfyEntity
 
+BLIND_DEVICE_CATAGORIES = {Category.INTERIOR_BLIND.value, Category.EXTERIOR_BLIND.value}
+SHUTTER_DEVICE_CATAGORIES = {Category.EXTERIOR_BLIND.value}
+SUPPORTED_CATAGORIES = {
+    Category.ROLLER_SHUTTER.value,
+    Category.INTERIOR_BLIND.value,
+    Category.EXTERIOR_BLIND.value,
+}
+
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the Somfy cover platform."""
 
     def get_covers():
         """Retrieve covers."""
-        categories = {
-            Category.ROLLER_SHUTTER.value,
-            Category.INTERIOR_BLIND.value,
-            Category.EXTERIOR_BLIND.value,
-        }
-
         devices = hass.data[DOMAIN][DEVICES]
 
         return [
             SomfyCover(
                 cover, hass.data[DOMAIN][API], hass.data[DOMAIN][CONF_OPTIMISTIC]
             )
             for cover in devices
-            if categories & set(cover.categories)
+            if SUPPORTED_CATAGORIES & set(cover.categories)
```

Comment:
nit: use categories instead of catagories

Reply:
fixed